src/strands_tools/mcp_client.py

```diff
@@ -0,0 +1,899 @@
+"""MCP Client Tool for Strands Agents.
+
+This tool provides a high-level interface for connecting to any MCP server
+with simplified configuration and enhanced functionality.
+
+It leverages the Strands SDK's MCPClient for robust connection management
+and implements a per-operation connection pattern for stability.
+"""
+
+import logging
+import time
+import uuid
+from dataclasses import dataclass
+from datetime import timedelta
+from threading import Lock
+from typing import Any, Dict, List, Optional
+
+from mcp import StdioServerParameters, stdio_client
+from mcp.client.sse import sse_client
+from mcp.client.streamable_http import streamablehttp_client
+from strands import tool
+from strands.tools.mcp import MCPClient, MCPTransport
+from strands.types.tools import AgentTool, ToolResult, ToolSpec, ToolUse
+
+logger = logging.getLogger(__name__)
+
+# Suppress warnings from MCP SDK about unknown notification types
+# This is particularly useful for test servers that send non-standard notifications
+# Users can override this by setting their own logging configuration
+logging.getLogger("mcp.shared.session").setLevel(logging.ERROR)
+
+# Default timeout for MCP operations
+DEFAULT_MCP_TIMEOUT = 30.0
```

Comment:
Can we get the default timeout from environment variable too?

STRANDS_MCP_TIMEOUT or similar 🤔